packages/simulation/src/convert.ts

```diff
@@ -0,0 +1,647 @@
+import { FSAGraphIn, FSAState, StateID, ReadSymbol } from './graph'
+import { AutomataState, AutomataTransition } from 'frontend/src/types/ProjectTypes'
+
+// EXTREMELY IMPORTANT NOTE:
+// The NFA to DFA conversion relies on sequential StateID's such that each state is ordered from 0 onwards without skipping a number.
+// Accounting for anything else gets extremely complicated, as such if non-sequential states can be formed, as such make sure
+// that reorder is called before working with this file
+
+// Essentially the graph cant be too heavily spaghetti (I.E states completely out of order and not in sequence, initial state in random
+// location, etc) or else this will not work. Perhaps some error handling could be done in the future but it will be quite difficult. I think
+// instead there should be a file that checks to ensure the flow of the graph makes logical sense before allowing it to do any
+// operations (future enhancement)
+
+// Define a type for the DFAGraph
+type DFAGraph = {
+  initialState: StateID;
+  states: AutomataState[];
+  transitions: AutomataTransition[];
+}
+
+// Graph traversal algorithm starting from the initial state that removes unreachable states and transitions
+export const removeUnreachableNFAStates = (nfaGraph: FSAGraphIn): FSAGraphIn => {
```

Comment:
Take a look at closureWithPredicate.ts, it basically implements the graph exploration you have here so you can use that instead of reinventing the wheel. Should work for `removeUnreachableDFAStates` also (or atleast should once I finish cleaning the types on it)

Reply:
I'll just leave it for now then if it isn't compatible with DFA graph as that's an important part of the algorithm